projects/extension/src/background/ConnectionManager.ts

```diff
@@ -215,13 +232,38 @@ export class ConnectionManager
     this.#client = undefined
   }
 
+  /**
+   * init Runs on onStartup and onInstalled of extension and
+   * connects the default relayChains to smoldot
+   */
+  init = () => {
```

Comment:
Also, IMO we should be calling `init` from the constructor. There's no reason it has to be a different function. If you don't want to initialize stuff, just don't create the `ConnectionManager`.

This would also make it possible to make the types of the various fields more precise. For example `#client` could be just of type `Client` instead of `Client | undefined`.

Reply:
`init` needs to be called in 2 different parts of the lifecycle of the extension. 
1 (existing) - on installation of the extension
2 (existing) - on startup of the browser
ofc the `init` can be called from the constructor instead of being a different function - though that will limit our control of the function based on the lifecycle (there is also the `onUpdateAvailable`, `onSuspend` and `onSuspendCanceled` that may be useful to have the init as a separate function. 

For now I will move this to the constructor - just mentioning the reasoning behind the init()

Reply:
If you want to "de-init" a `ConnectionManager` then you should instead use a variable whose type is `ConnectionManager | null` and switch it to `null`. The inside of the `ConnectionManager` shouldn't support a "not initialized" mode.